pkg/R/sparkR.R

```diff
@@ -36,14 +36,17 @@ sparkR.stop <- function(env) {
   if (exists(".sparkRjsc", envir = env)) {
     sc <- get(".sparkRjsc", envir = env)
     callJMethod(sc, "stop")
+    rm(".sparkRjsc", envir = env)
   }
 
```

Comment:
A rewrite of the two if statements could look like:
This also incorporates the Sys.sleep(1) within the if statement


  if (exists(".sparkRjsc", envir = env)&&exists(".sparkRCon", envir = env)) {
    cat("Stopping SparkR\n")
    sc <- get(".sparkRjsc", envir = env)
    callJMethod(sc, "stop")
    rm(".sparkRjsc", envir = env)

    callJStatic("SparkRHandler", "stopBackend")
    # Also close the connection and remove it from our env
    conn <- get(".sparkRCon", env)
    close(conn)
    rm(".sparkRCon", envir = env)
    # Finally, sleep for 1 sec to let backend finish exiting.
    # Without this we get port conflicts in RStudio when we try to 'Restart R'.
    Sys.sleep(1)
  }else{
    cat("SparkR Not Running: Call sparkR.init() to start SparkR")
  }